### ISSUE TITLE
Performance Problems

### ISSUE THREAD

**Issue by innominate227** [kaby76/AntlrVSIX#96]
Had to disable this extension as it caused major performance problems even when no ANTLR files were open.  I thought it was a R# issue at first, and used their tracing tool.  In the response you can see some Antlr calls are causing many seconds of UI blocking:  https://youtrack.jetbrains.com/issue/RSRP-481676

**Comment by kaby76** [comment]
The next release of Antlrvsix for VS2019 fixes this and many other performance issues. In this particular case, ithe server is reading every damn file in your source code directory. If you want, uninstall the current version of the extension, and grab the latest from the daily build area (see the instructions in the doc for the extension). An alternative is to use VSCode and the Antlrvsix extension for VSCode instead. The LSP client in VS2019, written by Microsoft, is years behind the what VSCode implements.

**Comment by kaby76** [comment]
Note to myself--I don't know why, but I took great care in mapping a file name into the proper case on disk within [FindDocument](https://github.com/kaby76/AntlrVSIX/blob/5fba2752ea797de42896511d2fc9b4d4bc792c7c/Workspaces/Document.cs#L77), which is done here twice because I push the Debug version to the Marketplace. It might be because I wanted to find the file if the user gave a bogus mix case file name that doesn't match what is on disk. So, this code may not be necessary. I'll need to mull this over. But, beside that, this code is a really crappy, which I yanked from StackOverflow. It constantly is reading the disk at each directory from root to the tail of the path. If I have to get the name with proper case (i.e., for Linux?), then memoize the arg and result.
